vowpalwabbit/kernel_svm.cc

```diff
@@ -322,22 +257,22 @@ void save_load_svm_model(svm_params& params, io_buf& model_file, bool read, bool
       model_file, reinterpret_cast<char*>(&(model->num_support)), sizeof(model->num_support), read, msg, text);
   // params.all->opts_n_args.trace_message<<"Read num support "<<model->num_support<< endl;
 
-  flat_example* fec = nullptr;
   if (read) { model->support_vec.reserve(model->num_support); }
 
   for (uint32_t i = 0; i < model->num_support; i++)
   {
     if (read)
     {
-      save_load_flat_example(model_file, read, fec);
-      svm_example* tmp = &calloc_or_throw<svm_example>();
-      tmp->init_svm_example(fec);
+      auto fec = VW::make_unique<flat_example>();
+      auto* tmp = &calloc_or_throw<svm_example>();
+      VW::model_utils::read_model_field(model_file, *fec, params.all->example_parser->lbl_parser);
+      tmp->ex = *fec;
       model->support_vec.push_back(tmp);
     }
     else
     {
-      fec = &(model->support_vec[i]->ex);
-      save_load_flat_example(model_file, read, fec);
+      VW::model_utils::write_model_field(model_file, model->support_vec[i]->ex, "_flat_example", false,
```

Comment:
Is the model binary output identical with the previous method?

Reply:
The previous method called `bin_write_fixed` / ``bin_read_fixed` on a flat example which resulted in a segfault. I made the new method analogous to how example is cached.